File: koma-core/src/main/java/org/koma/core/layout/KomaLayout.java
package org.koma.core.layout;

import lombok.Builder;
import lombok.Getter;
import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;
import org.koma.core.constant.ConfigConstant;

import java.nio.file.Path;

@Builder
@Getter
public class KomaLayout {
  public final Path basePath;

  public Path config() {
    return basePath.resolve(ConfigConstant.FILE_NAME);
  }

  @Contract(pure = true)
  public @NotNull Path content() {
    return basePath.resolve("content");
  }

  public @NotNull Path assets() {
    return basePath.resolve("assets");
  }

  public @NotNull Path templates() {
    return basePath.resolve("templates");
  }

  public @NotNull Path dist() {
    return basePath.resolve("dist");
  }
}

File: koma-compiler/src/main/kotlin/org/koma/core/model/KomaLayout.kt
package org.koma.core.model

import org.jetbrains.annotations.Contract
import org.koma.core.config.FILE_NAME
import java.nio.file.Path

data class KomaLayout(private val _basePath: Path) {
  val path: Path
    get() = _basePath

  fun config(): Path {
    return _basePath.resolve(FILE_NAME)
  }

  fun content(): Path {
    return _basePath.resolve("content")
  }

  fun assets(): Path {
    return _basePath.resolve("assets")
  }

  fun templates(): Path {
    return _basePath.resolve("templates")
  }

  fun dist(): Path {
    return _basePath.resolve("dist")
  }
}